Code/Framework/AzCore/AzCore/Asset/AssetSerializer.h

```diff
@@ -128,7 +128,7 @@ namespace AZ {
         {
         public:
             GenericClassGenericAsset()
-                : m_classData{ SerializeContext::ClassData::Create<ThisType>("Asset", GetSpecializedTypeId(), &m_factory, &AssetSerializer::s_serializer) }
+                : m_classData{ SerializeContext::ClassData::Create<ThisType>(AzTypeInfo<ThisType>::Name(), GetSpecializedTypeId(), &m_factory, &AssetSerializer::s_serializer) }
```

Comment:
The name doesn't affect ObjectStream serialized content. It is purely informational in that case.
For JSON Serialization content an `AzDeprecatedName<T>` has been specialized to register the old type name with the current UUID so that still works.